react-app/src/components/GamePage.js

```diff
@@ -21,6 +21,18 @@ function currentTime() {
   return new Date().getTime() / 1000.0;
 }
 
```

Comment:
How about _this_ for readability?  The task of "pulling one item out of a list" is moved into a function called extract.  The function itself is commented explaining what it does.  The code that calls it is fewer lines and less variables to keep track of, because the mechanical part is hidden in the function.  What do you think?